componenets/org.wso2.carbon.identity.application.authenticator.naver/src/main/java/org.wso2.carbon.identity.application.authenticator.naver/NaverCustomAuthenticator.java

```diff
@@ -159,4 +257,187 @@ public List<Property> getConfigurationProperties() {
+            authenticatorData.setPromptType(FrameworkConstants.AuthenticatorPromptType.REDIRECTION_PROMPT);
+            authenticatorData.setAdditionalData(getAdditionalData(context, false));
+        }
+        authenticatorData.setRequiredParams(requiredParameterList);
+
+        return Optional.of(authenticatorData);
+    }
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response, AuthenticationContext context) throws AuthenticationFailedException {
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("processAuthenticationResponse");
+        }
+
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            String clientId = (String)authenticatorProperties.get(CLIENT_ID);
+            String clientSecret = (String)authenticatorProperties.get(CLIENT_SECRET);
+            Boolean basicAuthEnabled = Boolean.parseBoolean((String)authenticatorProperties.get(IS_BASIC_AUTH_ENABLED));
+            String tokenEP = this.getTokenEndpoint(authenticatorProperties);
+            String token;
+            if (isTrustedTokenIssuer(context) && isNativeSDKBasedFederationCall(request)) {
+                String idToken = request.getParameter(ID_TOKEN_PARAM);
+                token = request.getParameter(NaverCustomAuthenticatorConstants.ACCESS_TOKEN_PARAM);
+                validateJWTToken(context, idToken);
+            } else {
+                String callbackUrl = getCallbackUrl(authenticatorProperties);
+                if (Boolean.parseBoolean((String) context.getProperty(NaverCustomAuthenticatorConstants.IS_API_BASED))) {
+                    callbackUrl = (String) context.getProperty(NaverCustomAuthenticatorConstants.REDIRECT_URL);
+                }
+                String code = getAuthorizationCode(request);
+                token = getToken(tokenEP, clientId, clientSecret, code, callbackUrl, basicAuthEnabled);
+            }
+
+            Boolean selfContainedTokenEnabled = Boolean.parseBoolean(authenticatorProperties
+                    .get(Oauth2GenericAuthenticatorConstants.SELF_CONTAINED_TOKEN_ENABLED));
+            String userInfo = getUserInfo(selfContainedTokenEnabled, token, authenticatorProperties);
+
+            this.buildClaims(context, userInfo);
+        } catch (ApplicationAuthenticatorException | MisconfigurationException e) {
+            logger.error("Failed to process Connect response.", e);
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        } catch (IdentityOAuth2Exception e) {
+            throw new AuthenticationFailedException("JWT token is invalid.");
+        } catch (ParseException | IdentityProviderManagementException | JOSEException e) {
+            throw new AuthenticationFailedException("JWT token validation Failed.", e);
+        }
+    }
+
+    private static AdditionalData getAdditionalData(
```

Comment:
Let's add method comments for more readability.

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-naver/pull/13/commits/545814df82608605708a7e15c0ac5b8368dd8507